### ISSUE TITLE
Model performance evaluation for superpixel localization

### ISSUE THREAD

**Issue by abdullah-al-masud** [NeelBhowmik/efficient-compact-fire-detection-cnn#1]
I have tried to use the inference scripts and fine tuned weight files. For, full frame classification, the accuracy can be calculated easily and I was able to evaluate the model performance properly. But for superpixel-localization, I am confused how to calculate model performance. I have already read the paper. 

After reading the paper, it seemed to me that the model's performance was evaluated in a classification manner. So, I tried to calculate the model accuracy under these conditions

Assumption: "If at least one superpixel is detected as fire, the image is classified as fire image"
Prediction: if prediction is 0, its classified as fire, if 1 then its classifier as nofire.
Test data-set: same data set mentioned in this repo, inside this directory "/fire-dataset-dunnings/images-224x224/test/"

But the performance I got didnt match the performance mentioned in the paper. I see too many false positives in the predicted images. I tried the same dataset mentioned in this repo. So, I guess my hypothesis of evaluation may be wrong.

Could you please suggest a way to evaluate model performance for superpixel localization method. I would like to get the model accuracy same as the paper stated.

**Comment by NeelBhowmik** [comment]
Hi,
Apology for the late reply. Did you able to solve the performance issue? Thanks.

**Comment by abdullah-al-masud** [comment]
No, I couldnt solve the issue. Can you please confirm the way to evaluate superpixel model's performance ?

For example, if I give an image and the SLIC algorithm divides it into 15 parts, among those 15 parts 3 are detected as fire by the model, then how do we evaluate the classification ?

**Comment by NeelBhowmik** [comment]
Let's consider, total superpixel (SP) = 15

fire (classID = 0) = 3 sp
no-fire (classID = 1) = 12 sp
`Ground Truth list of classID; gt = [0,0,0,1,1,1,1,1,1,1,1,1,1,1,1]`


Lets consider, model predicts 1st 3 no-fire sp as fire.
`model predict list of classID, pred = [0,0,0,0,0,0,1,1,1,1,1,1,1,1,1]`

To calculate classification related stats, you need to compare the above two lists (gt, pred). 
You can use any standard statistics library, such as _sklearn.metrics.classification_report_  
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html 

Thanks
